### ISSUE TITLE
Broken build system

### ISSUE THREAD

**Issue by retep998** [uutils/coreutils#655]
Your build system assumes that the package name and library name are the same for any given dependency, which is an invalid assumption.
Also, it is entirely not robust in that you have to specify all sub-dependencies in `deps.mk` which is just awful.
This results in issues like the following:
```
rustc -O -L /home/Peter/coreutils/build/ --extern aho_corasick=/home/Peter/coreutils/build/libaho_corasick.rli
b --extern bit_set=/home/Peter/coreutils/build/libbit_set.rlib --extern bit_vec=/home/Peter/coreutils/build/li
bbit_vec.rlib --extern crypto=/home/Peter/coreutils/build/libcrypto.rlib --extern getopts=/home/Peter/coreutil
s/build/libgetopts.rlib --extern kernel32_sys=/home/Peter/coreutils/build/libkernel32_sys.rlib --extern libc=/
home/Peter/coreutils/build/liblibc.rlib --extern memchr=/home/Peter/coreutils/build/libmemchr.rlib --extern nu
m_cpus=/home/Peter/coreutils/build/libnum_cpus.rlib --extern rand=/home/Peter/coreutils/build/librand.rlib --e
xtern regex=/home/Peter/coreutils/build/libregex.rlib --extern regex_syntax=/home/Peter/coreutils/build/libreg
ex_syntax.rlib --extern rustc_serialize=/home/Peter/coreutils/build/librustc_serialize.rlib --extern time=/hom
e/Peter/coreutils/build/libtime.rlib --extern unicode_width=/home/Peter/coreutils/build/libunicode_width.rlib
 --crate-type rlib --emit link,dep-info /home/Peter/coreutils/src/du/du.rs --out-dir /home/Peter/coreutils/bui
ld
C:/msys64/home/Peter/coreutils/src/du/du.rs:18:1: 18:19 error: can't find crate for `kernel32` which `time` de
pends on
C:/msys64/home/Peter/coreutils/src/du/du.rs:18 extern crate time;
                                               ^~~~~~~~~~~~~~~~~~
error: aborting due to previous error
```
If I add `kernel32-sys` to the `deps.mk` then `cp` gives me a warning and the same error occurs.
```
cp: cannot stat ‘/home/Peter/coreutils/deps/target/release/deps/libkernel32_sys-*.rlib’: No such file or directory
```
If I specify `kernel32` instead then this occurs
```
package id specification `kernel32` matched no packages
Makefile:304: recipe for target 'DEP_kernel32' failed
```
Please either fix your build system, or let me just use Cargo without mucking about in a maze of make.
Sincerely,
Someone on Windows trying to fix your code on Windows for you but blocked on this build system.

**Comment by ebfe** [comment]
I had a working cargo build system in #395, but it wasn't very popular. Unfortunately it bitrotted.

**Comment by nathanross** [comment]
fixed #715 and various commits on the build system by jbcrail since.
